libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java

```diff
@@ -265,161 +231,37 @@ public void onMilestoneTrigger(List<Milestone> triggeredMilestones, RouteProgres
    */
   @Override
   public void onUserOffRoute(Location location, boolean userOffRoute) {
-    if (userOffRoute) {
-      if (location.getTime() > timeIntervalSinceLastOffRoute
-        + TimeUnit.SECONDS.toMillis(mapboxNavigation.options().secondsBeforeReroute())) {
-        timeIntervalSinceLastOffRoute = location.getTime();
-        if (mapboxNavigation.getSessionState().eventLocation() == null) {
-          rerouteSessionsStateUpdate();
-        } else {
-          Point lastReroutePoint = Point.fromLngLat(
-            mapboxNavigation.getSessionState().eventLocation().getLongitude(),
-            mapboxNavigation.getSessionState().eventLocation().getLatitude());
-          if (TurfMeasurement.distance(lastReroutePoint,
-            Point.fromLngLat(location.getLongitude(), location.getLatitude()),
-            TurfConstants.UNIT_METERS)
-            > mapboxNavigation.options().minimumDistanceBeforeRerouting()) {
-            rerouteSessionsStateUpdate();
-          }
-        }
-      }
-    } else {
-      timeIntervalSinceLastOffRoute = location.getTime();
+    navigationQueueContainer.onUserOffRoute(location, userOffRoute);
+    recentDistancesFromManeuverInMeters.clear();
```

Comment:
What is `recentDistancesFromManeuverInMeters` and why does it need to be cleared here?

Reply:
@electrostat Yeah this is a good catch - once this is figured out, let's ship this

Reply:
Removed this clear. recentDistancesFromManeuverInMeters logs distances to upcoming maneuvers. If the clear is present then it never passes these values on to the rest of the system. Was present in the previous iteration of this system/service and the reason it was added. Removed now and all fixed.